src/activities/menu/Menu.qml

```diff
@@ -459,6 +500,131 @@ ActivityBase {
 
         }
 
+
+        Rectangle{
+            id: searchbar
+            width: horizontal ? parent.width/2 : parent.width - (section.width+10)
+            height: horizontal ? parent.height/10 : parent.height/17
+            visible:  menuActivity.currentTag === "search" ? true :false
+            anchors
+            {
+               bottom : horizontal ? activitiesGrid.top : undefined
+               top: horizontal ? undefined : parent.top
+               left: horizontal ? undefined : section.right
+            }
+            anchors.topMargin: horizontal ? undefined : 4
+            anchors.bottomMargin: horizontal ? undefined : 4
+            x: horizontal ? (parent.width/2) - (searchbar.width/2) : 0
+            opacity: 0.5
+            radius: 10
+            border.width: 2
+            border.color: "black"
+            gradient: Gradient {
+
+               GradientStop { position: 0.3; color: "#000" }
+               GradientStop { position: 0.9; color: "#666" }
+               GradientStop { position: 1.0; color: "#AAA" }
+
+           }
+
+            TextField{
+                id : input
+                anchors.fill: parent
+                text: ""
+                textColor: "black"
+                font.pixelSize: 28
```

Comment:
You must not use pixelSize, it does not scale properly on HDPI devices. Use fontSize instead like this:
font.pointSize: (18 + ApplicationSettings.baseFontSize) * ApplicationInfo.fontRatio